Applications/Spire/Source/SpireTester/AssociativeValueModelTester.cpp

```diff
@@ -0,0 +1,285 @@
+#include <doctest/doctest.h>
+#include <boost/optional/optional_io.hpp>
+#include "Spire/Ui/AssociativeValueModel.hpp"
+#include "Spire/Ui/LocalValueModel.hpp"
+
+using namespace Spire;
+
+TEST_SUITE("AssociativeValueModel") {
+  TEST_CASE("associate") {
+    auto model = AssociativeValueModel<std::string>();
+    auto bool_model1 = std::make_shared<LocalBooleanModel>(false);
+    auto bool_model2 = std::make_shared<LocalBooleanModel>(false);
+    model.associate(bool_model1, "model1");
+    model.associate(bool_model2, "model2");
+    bool_model1->set_current(true);
+    REQUIRE(bool_model1->get_current() == true);
+    REQUIRE(bool_model2->get_current() == false);
+    REQUIRE(model.get_current() == "model1");
+    bool_model2->set_current(true);
+    REQUIRE(bool_model1->get_current() == false);
+    REQUIRE(bool_model2->get_current() == true);
+    REQUIRE(model.get_current() == "model2");
+    bool_model2->set_current(false);
+    REQUIRE(bool_model1->get_current() == false);
+    REQUIRE(bool_model2->get_current() == true);
+    REQUIRE(model.get_current() == "model2");
+  }
```

Comment:
if I put the test code below at the end of the test case, it will not be passed. Based on my understanding, bool_model1->set_current(false) is not supposed to change the current of the model.
    bool_model1->set_current(false);
    REQUIRE(bool_model1->get_current() == false);
    REQUIRE(bool_model2->get_current() == true);
    REQUIRE(model.get_current() == "model2");

Reply:
Nice catch; there wasn't a condition for when the value isn't the current and is_selected is false, leading to the current value to be modified.